api/src/services/alert-service.ts

```diff
@@ -0,0 +1,81 @@
+import { IDBConnection } from '../database/db';
+import { IAlert, IAlertCreateObject, IAlertFilterObject, IAlertUpdateObject } from '../models/alert-view';
+import { AlertRepository } from '../repositories/alert-repository';
+import { DBService } from './db-service';
+
+export class AlertService extends DBService {
+  alertRepository: AlertRepository;
+
+  constructor(connection: IDBConnection) {
+    super(connection);
+
+    this.alertRepository = new AlertRepository(connection);
+  }
+
+  /**
+   * Get all alert records, including deactivated alerts
+   *
+   * @param {IAlertFilterObject} filterObject
+   * @return {*}  Promise<IAlert[]>
+   * @memberof AlertService
+   */
+  async getAlerts(filterObject: IAlertFilterObject): Promise<IAlert[]> {
+    return this.alertRepository.getAlerts(filterObject);
+  }
+
+  /**
+   * Get a specific alert by its ID
+   *
+   * @param {number} alertId
+   * @return {*}  Promise<IAlert>
+   * @memberof AlertService
+   */
+  async getAlertById(alertId: number): Promise<IAlert> {
+    return this.alertRepository.getAlertById(alertId);
+  }
+
+  /**
+   * Create and associate alert for survey.
```

Comment:
This mentions 'survey' but I dont think its survey specific, same with the below functions.